ios/Video/RCTVideo+ContextualActions.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  RCTVideo+ContextualActions.swift
+//  react-native-video
+//
+//  Created by Eduard Mazur on 14.01.2025.
+//
+
+import AVKit
+
+@available(tvOS 15.0, *)
+extension RCTVideo {
+    func configureContextualActions() {
+        guard let player = playerForContextualActions else {
+            print("Player is not initialized.")
+            return
+        }
+
+        let interval = CMTime(seconds: 0.5, preferredTimescale: CMTimeScale(NSEC_PER_SEC))
+
+        timeObserverToken = player.addPeriodicTimeObserver(forInterval: interval, queue: .main) { [weak self] currentTime in
+            self?.updateContextualActions(currentTime: currentTime)
+        }
+    }
+
+    func removeContextualActionsTimeObserver() {
+        guard let player = playerForContextualActions, let token = timeObserverToken else { return }
+        player.removeTimeObserver(token)
+        timeObserverToken = nil
+    }
+
+    func updateContextualActions(currentTime: CMTime) {
+        let currentSeconds = CMTimeGetSeconds(currentTime)
+
+        guard let playerViewController = playerViewControllerForContextualActions else {
+            return
+        }
+        
+        for actionData in contextualActionData {
+            if currentSeconds >= actionData.startAt,
+               let endAt = actionData.endAt, currentSeconds < endAt {
+                handleAction(actionData, playerViewController: playerViewController)
+                return
+            } else if currentSeconds >= actionData.startAt, actionData.endAt == nil {
+                handleAction(actionData, playerViewController: playerViewController)
+                return
+            }
+        }
+
+        if currentContextualState != .none {
+            playerViewController.contextualActions = []
+            currentContextualState = .none
+        }
+    }
+
+    func handleAction(_ actionData: ContextualActionData, playerViewController: AVPlayerViewController) {
+        switch actionData.action {
+        case "SkipIntro":
+            if currentContextualState != .skipIntro {
+                let skipIntroAction = UIAction(title: "Skip Intro") { [weak self] _ in
```

Comment:
it would be great to make these text configurable for internationnalization support ! 
"Skip Intro" => "Skip intro" by default I think 
and "Next Episode"